src/interfaces/vault.ts

```diff
@@ -330,11 +330,16 @@ export class VaultInterface<T extends ChainId> extends ServiceInterface<T> {
       if (token === EthAddress) {
         throw new SdkError("deposit:v2:eth not implemented");
       } else {
-        const vaultContract = new Contract(vault, VaultAbi, signer);
+        const shouldUserPartner = this.yearn.services.partner && this.yearn.services.partner.isAllowed(vault);
+        const vaultContract =
+          (shouldUserPartner && this.yearn.services.partner) || new Contract(vault, VaultAbi, signer);
```

Comment:
Lets try to keep types not mixed for same constant. Here `vaultContract` is either a partner service instance or a contract instance, which might be confusing